lib/iopromise/deferred/promise.rb

```diff
@@ -5,10 +5,14 @@
 module IOPromise
   module Deferred
     class DeferredPromise < ::IOPromise::Base
-      def initialize(&block)
+      def initialize(timeout = nil, &block)
```

Comment:
It might make things easier for ourselves if we go with a keyword argument here.

```ruby
IOPromise::Deferred.new(timeout: 5) { do_the_things }
# versus
IOPromise::Deferred.new(5) { do_the_things }
```

Reply:
That's a great idea, much clearer :+1: